### PR TITLE
Adding a dashboard defaults wrapper component

### PR DESCRIPTION
**Pull request by AdityaHegde**
## Checklist
- [x] Manual verification
- [x] Unit test coverage
- [x] E2E test coverage
- [x] Needs manual QA?

#### Issue addressed: 
We have a lot of adhoc code to make sure defaults are set for a dashboard store.

#### Details:
Adding a wrapper component that initialised a dashboard store. Also adding `init` method and removing all init code from `sync` and `syncFromUrl`.

## Steps to Verify

### CODE REVIEW

**Review by skokenes — APPROVED** [review]
Looks great, the navigation between dashboards with different states seems much better now!
While the code looked good, I didn't totally follow how this new initialization works in with URL syncing to prevent the bugs we saw before. Can you provide some more explanation?
Also, if you haven't already, can you be sure to test this on a local cloud dev environment before merging?

File: web-common/src/features/dashboards/DashboardStateProvider.svelte [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      );
+    }
+  }
+
+  $: if ($metaQuery.data && $timeRangeQuery.data) {
+    syncDashboardState();
+  }
+</script>
+
+<slot />
```
[the remark this answers is not in the document]
Comment (AdityaHegde) [reply web-common/src/features/dashboards/DashboardStateProvider.svelte:34]: Yes. I can add that.

File: web-common/src/features/dashboards/proto-state/dashboard-url-state.ts [hunk 2]
```diff
@@ -69,8 +69,17 @@ export function useDashboardUrlSync(
...
   return dashboardUrlState.subscribe((state) => {
     if (state.urlName !== metricViewName) return;
+    if (!state.proto) {
+      isInit = true;
+      return;
+    }
+    if (isInit && state.proto === state.defaultProto) {
+      lastKnownProto = state.proto;
+      isInit = false;
+    }
```
[the remark this answers is not in the document]
Comment (AdityaHegde) [reply web-common/src/features/dashboards/proto-state/dashboard-url-state.ts:82]: `state.proto` is not defined when dashboard state is not created. When it is created we set `defaultProto` to `proto` which goes into the if branch.

**Comment by AdityaHegde** [comment]
Added some comments in the code to clarify the code. The test failures are unreated E2E instability, they pass locally.

File: web-common/src/features/dashboards/proto-state/dashboard-url-state.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (AdityaHegde) [reply web-common/src/features/dashboards/proto-state/dashboard-url-state.ts]: Updated the logic to make it simpler

**Review by skokenes — APPROVED** [review]
Awesome! LGTM

**Pull request merged**